src/modules/sync/lib/convert-tree-to-sync-options.ts

```diff
@@ -6,81 +6,100 @@ import type { SyncOption } from 'src/types';
 type PushOptionsWithSelections = {
 	optionsToSync: SyncOption[];
 	specificSelections?: {
-		plugins?: string[];
-		themes?: string[];
-		uploads?: string[];
+		paths?: string[];
 	};
 };
 
-const isSyncOption = ( value: string ): value is SyncOption => {
-	return Object.keys( SYNC_OPTIONS ).includes( value );
+const iterateOverCheckedNodes = (
+	nodes: TreeNode[] | undefined,
+	visit: ( node: TreeNode ) => void
+): void => {
+	if ( ! nodes?.length ) return;
+	for ( const node of nodes ) {
+		if ( node.checked ) {
+			visit( node );
+		} else if ( node.indeterminate && node.children?.length ) {
+			iterateOverCheckedNodes( node.children, visit );
+		}
+	}
+};
+
+const collectPathIds = ( nodes: TreeNode[] | undefined ): string[] => {
+	const out: string[] = [];
+	iterateOverCheckedNodes( nodes, ( node ) => {
+		if ( node.pathId ) out.push( node.pathId );
+	} );
+	return out;
 };
 
-const collectPathIds = ( nodes: TreeNode[], pathIds: string[] = [] ): string[] => {
-	nodes.forEach( ( node ) => {
-		if ( node.checked && node.pathId ) {
-			pathIds.push( node.pathId );
-		} else if ( node.indeterminate && node.children ) {
-			collectPathIds( node.children, pathIds );
+const convertTreeToSyncCategories = (
+	nodes: TreeNode[] | undefined
+): { paths: string[]; options: SyncOption[] } => {
+	const paths = new Set< string >();
+	const options = new Set< SyncOption >();
+
+	iterateOverCheckedNodes( nodes, ( node ) => {
+		if ( ! node.path ) return;
+
+		const p = node.path.replace( /^\/?wp-content\//, '' );
```

Comment:
```suggestion
		const nodePath = node.path.replace( /^\/?wp-content\//, '' );
```
Nit, but there's no reason not to use descriptive variable names.

Reply:
https://github.com/Automattic/studio/pull/1940/commits/5db98b7b459df0d147a4c0d3a4bc1bab6ee4987a